lib/src/signed_video_h26x_auth.c

```diff
@@ -1065,6 +1065,25 @@ signed_video_add_h26x_nalu(signed_video_t *self, const uint8_t *nalu_data, size_
         DEBUG_LOG("No cryptographic information found in SEI. Using default hash algo");
         self->validation_flags.hash_algo_known = true;
       }
+      // If the NALU is golden SEI, proceed with the validation process.
+      if (nalu.is_golden_sei) {
+        prepare_for_validation(self);
+        // Check the status of the verified signature hash for the GOP info.
+        prepare_for_validation(self);
+        switch (self->gop_info->verified_signature_hash) {
```

Comment:
This switch is a copy-paste from another part of the code. Then I suggest to make a function for it.